presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PruneCrossJoinColumns.java

```diff
@@ -63,8 +63,6 @@ public Pattern getPattern()
                 .map(ImmutableSet::copyOf)
                 .map(dependencies ->
                         parent.replaceChildren(ImmutableList.of(
-                                joinNode.replaceChildren(ImmutableList.of(
-                                        restrictOutputs(idAllocator, joinNode.getLeft(), dependencies).orElse(joinNode.getLeft()),
-                                        restrictOutputs(idAllocator, joinNode.getRight(), dependencies).orElse(joinNode.getRight()))))));
+                                restrictChildOutputs(idAllocator, joinNode, dependencies, dependencies).get())));
```

Comment:
It won't, because cross joins pass through all their inputs.  So one of the children will have been pruned, if the outputs of the join can be pruned.